gitops/common/app.py

```diff
@@ -28,6 +28,7 @@ def __init__(
         deployments = deployments or {}
         secrets = secrets or {}
         self.values = self._make_values(deployments, secrets)
+        self.namespace: str = self.values["namespace"]
```

Comment:
Do we want this here, or do we want to add an `@property` like we've done for the other attribs?
My preference would be for the latter, to:
- avoid storing the same info in multiple places.
- prevent users from setting `namespace` directly on the app, and expecting it to go somewhere.

Reply:
I want it to be a declared value. namespace lives in the yaml but should be a thing that exists for all.

Namespace doesn't actually propagate, we don't use namespace in the deployment values at all; we only use it in the `helm install --namespace` and thats why it doesn't belong in values.